statistics/analyze_jobs.go

```diff
@@ -15,128 +15,33 @@
 package statistics
 
 import (
-	"sort"
 	"sync"
 	"time"
 )
 
-type analyzeJobs struct {
-	sync.Mutex
-	jobs    map[*AnalyzeJob]struct{}
-	history []*AnalyzeJob
-}
-
-var analyzeStatus = analyzeJobs{jobs: make(map[*AnalyzeJob]struct{}), history: make([]*AnalyzeJob, 0, numMaxHistoryJobs)}
-
 // AnalyzeJob is used to represent the status of one analyze job.
 type AnalyzeJob struct {
-	sync.Mutex
+	ID            *uint64
 	DBName        string
 	TableName     string
 	PartitionName string
 	JobInfo       string
-	RowCount      int64
 	StartTime     time.Time
 	EndTime       time.Time
-	State         string
-	updateTime    time.Time
+	Delta         struct {
+		sync.Mutex
+		Count        int64
+		LastDumpTime time.Time
+	}
```

Comment:
This implementation seems dangerous and hard to maintain since it needs the user(who access and modify Delta) to lock it manually.
I suggest implementing it as a dedicated and separate structure and hiding the lock in its methods:
```
type AnalyzeDelta struct {...}

func (d *AnalyzeDelta) Inc(cnt int64) {lock(); defer unlock(); d.Count+=cnt}
func (d *AnalyzeDelta) Count() int64 {rlock(); defer runlock(); return d.Count}
```

Reply:
Done.